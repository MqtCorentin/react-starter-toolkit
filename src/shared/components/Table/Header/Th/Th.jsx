import React from 'react';
import PropTypes from 'prop-types';
import WithClassNameModifier from 'shared/helpers/WithClassNameModifier';

const Th = WithClassNameModifier(({ className, children, ariaLabel, ...rest }) => (
  <th className={className} aria-label={ariaLabel} {...rest}>
    {children}
  </th>
));

Th.propTypes = {
  ariaLabel: PropTypes.string,
  role: PropTypes.string,
  className: PropTypes.string,
};

Th.defaultProps = {
  ariaLabel: 'table-body-line',
  role: 'columnheader',
  className: 'af-table__th',
};

export default Th;
