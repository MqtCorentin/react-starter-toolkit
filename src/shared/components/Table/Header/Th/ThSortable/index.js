export { default } from './ThSortable.container';