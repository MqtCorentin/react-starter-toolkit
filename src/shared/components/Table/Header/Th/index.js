export { default } from './Th.container';
