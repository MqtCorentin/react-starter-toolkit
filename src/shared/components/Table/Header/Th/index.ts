export { default } from './Th.container';
export type { TThContainer } from './Th.container';
export type { TsetSort } from './ThSortable';
