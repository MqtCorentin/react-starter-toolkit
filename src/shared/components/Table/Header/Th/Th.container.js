import React from 'react';
import { string } from 'prop-types';
import ThSortable from './ThSortable';
import Th from './Th';

export const ThEnhanced = ({ field, sorting, onSort, ...rest }) =>
  field ? <ThSortable {...rest} sorting={sorting} onSort={onSort} field={field} /> : <Th {...rest} />;

const ThEnhancedPropTypes = {
  field: string,
};

const ThEnhancedDefaultProps = {
  field: '',
};

ThEnhanced.propTypes = ThEnhancedPropTypes;
ThEnhanced.defaultProps = ThEnhancedDefaultProps;

export default ThEnhanced;
