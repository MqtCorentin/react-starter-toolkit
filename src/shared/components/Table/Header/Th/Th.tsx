import { ReactNode } from 'react';
import { default as TableTk } from '@axa-fr/react-toolkit-table';
import ThSortable, { TsetSort } from './ThSortable';

export type TTh = Omit<TsetSort, 'onSort' | 'field'> & {
  ThSortableCmpt?: typeof ThSortable;
  ThCmpt?: typeof TableTk.Th;
  children?: ReactNode;
  classModifier?: string;
  onSort?: TsetSort['onSort'];
  field?: TsetSort['field'];
};

const Th = ({ field, sorting, onSort, ThSortableCmpt = ThSortable, ThCmpt = TableTk.Th, ...rest }: TTh) =>
  field && onSort ? <ThSortableCmpt {...rest} sorting={sorting} onSort={onSort} field={field} /> : <ThCmpt {...rest} />;

export default Th;
