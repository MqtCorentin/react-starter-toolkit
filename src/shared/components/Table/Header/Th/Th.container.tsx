import { ReactNode } from 'react';
import ThSortable, { TsetSort } from './ThSortable';
import Th from './Th';

export type TThContainer = Omit<TsetSort, 'onSort' | 'field'> & {
  ThSortableCmpt?: typeof ThSortable;
  ThCmpt?: typeof Th;
  children?: ReactNode;
  classModifier?: string;
  onSort?: TsetSort['onSort'];
  field?: TsetSort['field'];
};

export const ThContainer = ({ field, sorting, onSort, ThSortableCmpt = ThSortable, ThCmpt = Th, ...rest }: TThContainer) =>
  field && onSort ? <ThSortableCmpt {...rest} sorting={sorting} onSort={onSort} field={field} /> : <ThCmpt {...rest} />;

export default ThContainer;
