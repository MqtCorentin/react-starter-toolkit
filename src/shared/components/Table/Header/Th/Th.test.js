import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import Th from './Th';

const defaultProps = {
  children: null,
};
const container = document.createElement('tr');

const renderWithContainer = Component =>
  render(<Component />, {
    container: document.body.appendChild(container),
  });

describe('Th', () => {
  it('Render <Th/>', () => {
    const { asFragment } = renderWithContainer(() => <Th {...defaultProps} >child th</Th>);
    expect(asFragment()).toMatchSnapshot();
  });
  
});
