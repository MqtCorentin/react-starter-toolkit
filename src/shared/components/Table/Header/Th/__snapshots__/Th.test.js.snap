// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Th Render <Th/> 1`] = `
<DocumentFragment>
  child th
</DocumentFragment>
`;
